### YamlMime:ManagedReference
items:
- uid: System.Activities.UnhandledExceptionAction
  id: UnhandledExceptionAction
  children:
  - System.Activities.UnhandledExceptionAction.Abort
  - System.Activities.UnhandledExceptionAction.Cancel
  - System.Activities.UnhandledExceptionAction.Terminate
  langs:
  - csharp
  name: UnhandledExceptionAction
  nameWithType: UnhandledExceptionAction
  fullName: System.Activities.UnhandledExceptionAction
  type: Enum
  summary: "Указывает действие, которое возникает, когда исключение покидает корневой элемент рабочего процесса."
  remarks: "<xref:System.Activities.WorkflowApplication.OnUnhandledException%2A>Функция вызывается в том случае, если исключение покидает корневой рабочего процесса.</xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs>Предоставляет доступ к исключению, а также указатель на <xref:System.Activities.Activity>, вызвавшего исключение.</xref:System.Activities.Activity> </xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> Завершить является действием по умолчанию, если не <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A>указан обработчик.</xref:System.Activities.WorkflowApplication.OnUnhandledException%2A>"
  example:
  - "The following example invokes a workflow that throws an exception. The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked. The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is terminated.  \n  \n [!code-cs[CFX_WorkflowApplicationExample#6](~/add/codesnippet/csharp/t-system.activities.unha_1.cs)]"
  syntax:
    content: public enum UnhandledExceptionAction
  inheritance:
  - System.Enum
  platform:
  - net462
- uid: System.Activities.UnhandledExceptionAction.Abort
  id: Abort
  parent: System.Activities.UnhandledExceptionAction
  langs:
  - csharp
  name: Abort
  nameWithType: UnhandledExceptionAction.Abort
  fullName: System.Activities.UnhandledExceptionAction.Abort
  type: Field
  assemblies:
  - System.Activities
  namespace: System.Activities
  summary: "Указывает, что <xref href=&quot;System.Activities.WorkflowApplication&quot;> </xref> должна прервать выполнение рабочего процесса."
  remarks: "В результате <xref:System.Activities.WorkflowApplication.Aborted%2A>после прерывания процесса вызывается.</xref:System.Activities.WorkflowApplication.Aborted%2A> Необработанное исключение служит причиной прерывания."
  example:
  - "The following example invokes a workflow that throws an exception. The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked. The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is aborted.  \n  \n [!code-cs[CFX_WorkflowApplicationExample#7](~/add/codesnippet/csharp/t-system.activities.unha_2.cs)]"
  syntax:
    content: Abort
    return:
      type: System.Activities.UnhandledExceptionAction
      description: "Для добавления."
  exceptions: []
  platform:
  - net462
- uid: System.Activities.UnhandledExceptionAction.Cancel
  id: Cancel
  parent: System.Activities.UnhandledExceptionAction
  langs:
  - csharp
  name: Cancel
  nameWithType: UnhandledExceptionAction.Cancel
  fullName: System.Activities.UnhandledExceptionAction.Cancel
  type: Field
  assemblies:
  - System.Activities
  namespace: System.Activities
  summary: "Указывает, что <xref href=&quot;System.Activities.WorkflowApplication&quot;> </xref> должен запланировать отмену корневого действия и возобновить выполнение."
  remarks: "В результате <xref:System.Activities.WorkflowApplication.Completed%2A>после завершения процесса отмены вызывается.</xref:System.Activities.WorkflowApplication.Completed%2A>"
  example:
  - "The following example invokes a workflow that throws an exception. The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked. The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is cancelled.  \n  \n [!code-cs[CFX_WorkflowApplicationExample#8](~/add/codesnippet/csharp/t-system.activities.unha_3.cs)]"
  syntax:
    content: Cancel
    return:
      type: System.Activities.UnhandledExceptionAction
      description: "Для добавления."
  exceptions: []
  platform:
  - net462
- uid: System.Activities.UnhandledExceptionAction.Terminate
  id: Terminate
  parent: System.Activities.UnhandledExceptionAction
  langs:
  - csharp
  name: Terminate
  nameWithType: UnhandledExceptionAction.Terminate
  fullName: System.Activities.UnhandledExceptionAction.Terminate
  type: Field
  assemblies:
  - System.Activities
  namespace: System.Activities
  summary: "Указывает, что <xref href=&quot;System.Activities.WorkflowApplication&quot;> </xref> должен запланировать завершение корневого действия и возобновить выполнение."
  remarks: "В результате <xref:System.Activities.WorkflowApplication.Completed%2A>после завершения процесса вызывается.</xref:System.Activities.WorkflowApplication.Completed%2A> Необработанное исключение служит причиной завершения. Завершить является действием по умолчанию, если не <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A>указан обработчик.</xref:System.Activities.WorkflowApplication.OnUnhandledException%2A>"
  example:
  - "The following example invokes a workflow that throws an exception. The exception is unhandled by the workflow and the <xref:System.Activities.WorkflowApplication.OnUnhandledException%2A> handler is invoked. The <xref:System.Activities.WorkflowApplicationUnhandledExceptionEventArgs> are inspected to provide information about the exception, and the workflow is terminated.  \n  \n [!code-cs[CFX_WorkflowApplicationExample#6](~/add/codesnippet/csharp/t-system.activities.unha_1.cs)]"
  syntax:
    content: Terminate
    return:
      type: System.Activities.UnhandledExceptionAction
      description: "Для добавления."
  exceptions: []
  platform:
  - net462
references:
- uid: System.Enum
  isExternal: false
  name: System.Enum
- uid: System.Activities.UnhandledExceptionAction.Abort
  parent: System.Activities.UnhandledExceptionAction
  isExternal: false
  name: Abort
  nameWithType: UnhandledExceptionAction.Abort
  fullName: System.Activities.UnhandledExceptionAction.Abort
- uid: System.Activities.UnhandledExceptionAction
  parent: System.Activities
  isExternal: false
  name: UnhandledExceptionAction
  nameWithType: UnhandledExceptionAction
  fullName: System.Activities.UnhandledExceptionAction
- uid: System.Activities.UnhandledExceptionAction.Cancel
  parent: System.Activities.UnhandledExceptionAction
  isExternal: false
  name: Cancel
  nameWithType: UnhandledExceptionAction.Cancel
  fullName: System.Activities.UnhandledExceptionAction.Cancel
- uid: System.Activities.UnhandledExceptionAction.Terminate
  parent: System.Activities.UnhandledExceptionAction
  isExternal: false
  name: Terminate
  nameWithType: UnhandledExceptionAction.Terminate
  fullName: System.Activities.UnhandledExceptionAction.Terminate
